packages/renderers/renderer-react/server.js

```diff
@@ -26,16 +26,24 @@ function check(Component, props, children) {
     return h('div');
   }
 
-  renderToStaticMarkup(Tester, props, children, {});
+  await renderToStaticMarkup(Tester, props, children, {});
 
   if (error) {
     throw error;
   }
+
   return isReactComponent;
 }
 
-function renderToStaticMarkup(Component, props, children, metadata) {
-  const vnode = h(Component, { ...props, children: h(StaticHtml, { value: children }), innerHTML: children });
+async function renderToStaticMarkup(Component, props, children, metadata) {
+  const childrenValue = children || (await props.children);
```

Comment:
Is `props.children` async here because it triggers another render call? Does Astro generally resolve `Promise` props, or is this a special case? I’m asking mostly out of curiosity, but I could also see this being quite surprising, both as a user and as a renderer author. It might be better to expose an API that allows renderers to supplement `children`, and hide the async nature of that as an Astro implementation detail.

Reply:
It's async because Astro renders everything async. It doesn't trigger another render call, it's just that props.children in this case (because it's passed as a prop is `h('div', 'prop')` and our h() function returns a promise.

You're right that this could be less than ideal in some cases, but remember that this only comes into play when you are passing Astro HTML to a framework component, something that's already kind of a weird thing to do. Mostly framework components deal with their own component/vnode types. 

I don't think we should await all props or anything like that, there's legit use cases for passing promises into components.

Reply:
Where I see this being an issue in React and similar frameworks is with `Suspense`, and more generally with prop drilling scenarios around data fetching. IMO renderers should be able to receive props and pass them to components the same way they would in their own runtime, and the async nature of Astro rendering handled by Astro.

This should be achievable with Astro’s `h` function already, because it’s just a function—calls are already executed from the innermost expression out to the outermost. If that’s too complex for whatever reason, it might make sense to return a subclass of `Promise` so renderers can at least distinguish them from cases where a `Promise` was passed intentionally. (I realize that’s probably unusual for `children`, but it can and does happen.)

Reply:
IMO this is a good callout. We should let the renderer handle it if `children` is a Promise other than what Astro's `h` function generates.

Other than that, can we normalize `props.children` and `children` _before_ passing them down to renderers? This logic would likely move to `__astro_component` and be handled like this:

```ts
const childrenValue = children || (await props.children);
const { html } = await __renderer.renderToStaticMarkup(Component, props, childrenValue, ...)
```

Reply:
The special `children` property is specific to React and Preact. That prop is just another prop in other frameworks and we shouldn't mess with a user's props in general, so the logic for handling it belongs in the React / Preact renderer.

Reply:
Actually now I'm thinking we shouldn't support this. Astro is not React. You pass children in Astro files via content, not props.